### PR TITLE
WIP: Upgrade Starscream from 3.x -> 4.x

### PR DESCRIPTION
**Pull request by danielrbrowne**
This PR updates the version of Starscream in use from 3.x to 4.x

This addresses #277 (and will allow other Websocket related issues to be addressed in the near future).

**NOTES:**
- This PR is currently WIP (and therefore will not pass CI) until https://github.com/daltoniam/Starscream/issues/811 is resolved upstream in Starscream.

### CODE REVIEW

File: Sources/PusherSwift.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ let CLIENT_NAME = "pusher-websocket-swift"
     public init(key: String, options: PusherClientOptions = PusherClientOptions()) {
         self.key = key
         let urlString = constructUrl(key: key, options: options)
-        let ws = WebSocket(url: URL(string: urlString)!)
+        let ws = WebSocket(request: URLRequest(url: URL(string: urlString)!))
```
Comment (JonathanDowning) [inline Sources/PusherSwift.swift:29]: Hey @danielrbrowne, just a curiosity question.

Have you experimented with using Starscream's native support for `URLSessionWebSocketTask`?

It's enabled by changing this line to 
``` swift
let ws = WebSocket(request: URLRequest(url: URL(string: urlString)!), useCustomEngine: false)
```

It used to be enabled by default but now is an opt-in. It's not clear why this change was made.
However in my testing (albeit limited), the native socket implementation seemed to work rather well.
Would you be opposed to testing the native iOS sockets implementation? 
I don't know about you, but if I was given the choice of using native networking code versus third party, I'd use native every time! 

Hope you're staying safe! Jon

**Comment by danielrbrowne** [comment]
It is likely that this PR will be closed in favour of #289 instead. This should simplify our dependency graph going forward, with the caveat that the change to native WebSockets will necessitate a version bump to 9.x since deployment targets will have to be increased several versions to allow use of the native API. @JonathanDowning this may be relevant to your interests.

File: Sources/PusherSwift.swift (hunk 1 above)
Comment (danielrbrowne) [reply Sources/PusherSwift.swift:29]: @JonathanDowning The reasoning for avoiding this until now was the fact that this SDK had minimum deployment targets which wouldn't allow for the use of Starscream's support of `URLSessionWebSocketTask`. We have now re-evaluated this blocker, see [this comment](https://github.com/pusher/pusher-websocket-swift/pull/280#issuecomment-696609804) for more info.

**Comment by danielrbrowne** [comment]
Closing for reasons above.

**Pull request closed without merging**